flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/TimestampsAndWatermarksOperator.java

```diff
@@ -83,12 +81,14 @@ public void open() throws Exception {
 		super.open();
 
 		timestampAssigner = watermarkStrategy.createTimestampAssigner(this::getMetricGroup);
-		watermarkGenerator = watermarkStrategy.createWatermarkGenerator(this::getMetricGroup);
+		watermarkGenerator = new NoOpWatermarkGenerator<>(
```

Comment:
Here I would do:
```
if (emitProgressiveWatermarks) {
  watermarkGenerator = watermarkStrategy.createWatermarkGenerator(this::getMetricGroup);
} else {
  watermarkGenerator = new NoOpWatermarkGenerator();
}
```

Reply:
@dawidwys We could reuse `NoWatermarksGenerator` instead of `NoOpWatermarkGenerator `.

Reply:
Yes! I wasn't aware of that class.